ctl/install/install.go

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package install
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+
+	"kmesh.net/kmesh/ctl/utils"
+)
+
+type GitHubFile struct {
+	Name        string `json:"name"`
+	DownloadURL string `json:"download_url"`
+}
+
+func NewCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "install",
+		Short: "install Kmesh with all the resources",
+		Example: `# Install all Kmesh resources (defaults to main):
+kmeshctl install
+
+# Install a specific version of kmesh
+kmeshctl install --version 0.5`,
+		Run: func(cmd *cobra.Command, args []string) {
+			version, err := cmd.Flags().GetString("version")
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			cli, err := utils.CreateKubeClient()
+			if err != nil {
+				log.Fatalf("failed to create cli client: %v", err)
+				os.Exit(1)
+			}
+
+			fmt.Println("install kmesh version: ", version)
+
+			combinedYAMLFile := getYAMLFile(version)
+			err = cli.ApplyYAMLContents("", combinedYAMLFile)
+			if err != nil {
+				log.Fatal(err)
+			}
+		},
+	}
+
+	cmd.Flags().String("version", "main", "Version of the resources to initialize")
+	return cmd
+}
+
+func getYAMLFile(version string) string {
+	var url string
+	if version != "main" {
+		url = fmt.Sprintf("https://api.github.com/repos/kmesh-net/kmesh/contents/deploy/yaml?ref=release-%s", version)
```

Comment:
I donot think this is quite correct, because?:
1.  we didn't update the image tag 
2. should not use the github branch, but use the tag instead

So in short should we use helm chart https://github.com/kmesh-net/kmesh/releases/download/v0.5.0/kmesh-helm-v0.5.0.tgz 

or the OCI chart image https://github.com/orgs/kmesh-net/packages/container/package/kmesh-helm

Reply:
It is using release tag? Using helm will introduce new dependencies which is totally unnecessary.

Reply:
See the released version above